npe2/_from_npe1.py

```diff
@@ -39,28 +38,42 @@
 
 NPE1_EP = "napari.plugin"
 NPE2_EP = "napari.manifest"
+NPE1_IMPL_TAG = "napari_impl"  # same as HookImplementation.format_tag("napari")
+
+
+class HookImplementation:
+    def __init__(
+        self,
+        function: Callable,
+        plugin: Optional[ModuleType] = None,
+        plugin_name: Optional[str] = None,
+        **kwargs,
+    ):
+        self.function = function
+        self.plugin = plugin
+        self.plugin_name = plugin_name
+        self._specname = kwargs.get("specname")
```

Comment:
Hum, why not:
```suggestion
        *,
        specname=None
    ):
        self.function = function
        self.plugin = plugin
        self.plugin_name = plugin_name
        self._specname = specname
```

Are you afraid people will see specname in the signature ? 
(we can also keep `**kwargs`, if you want to allow passing more arguments. )